bundles/org.openhab.binding.ondilo/src/main/java/org/openhab/binding/ondilo/internal/OndiloBridge.java

```diff
@@ -100,28 +104,32 @@ public synchronized void pollOndiloICOs() {
         try {
             OndiloApiClient apiClient = this.apiClient;
             if (apiClient != null) {
+                String userInfoJson = apiClient.get("/user/info");
+                logger.trace("userInfo: {}", userInfoJson);
+                // Parse JSON to DTO
+                Gson gson = new Gson();
+                UserInfo userInfo = gson.fromJson(userInfoJson, new TypeToken<UserInfo>() {
```

Comment:
It would be an improvement if the json parsing is done at the apiClient. Cleaner and de-duplication of code.

Possible signature:
``` 
public T get("/user/info", T type);
```
I would aso suggest to move the synchronized to the apiclient.

Reply:
> I would aso suggest to move the synchronized to the apiclient.

`handleCommand()` and `scheduler.scheduleWithFixedDelay()` can still call `pollOndiloICOs()` in parallel

Reply:
What object is this? It seems like you want to parse it to something link List<String> but it is actually a String[]

Reply:
error resolved